bridges/TwitterBridge.php

```diff
@@ -251,94 +312,145 @@ public function collectData(){
 			}
 		}
 
+		// Filter out unwanted tweets
+		foreach ($data as $tweet) {
+			// Filter out retweets to remove possible duplicates of original tweet
+			switch($this->queriedContext) {
+			case 'By keyword or hashtag':
+				if (isset($tweet->retweeted_status) && substr($tweet->full_text, 0, 4) === 'RT @') {
+					continue 2;
+				}
+				break;
+			}
+			$tweets[] = $tweet;
+		}
+
 		$hidePictures = $this->getInput('nopic');
 
-		$promotedTweetIds = array_reduce($data->timeline->instructions[0]->addEntries->entries, function($carry, $entry) {
-			if (!isset($entry->content->item)) {
-				return $carry;
-			}
-			$tweet = $entry->content->item->content->tweet;
-			if (isset($tweet->promotedMetadata)) {
-				$carry[] = $tweet->id;
-			}
-			return $carry;
-		}, array());
+		// $promotedTweetIds = array_reduce($data->timeline->instructions[0]->addEntries->entries, function($carry, $entry) {
+		//	if (!isset($entry->content->item)) {
+		//		return $carry;
```

Comment:
Instead of commenting lines, you should remove them. Better code quality.

Reply:
You are right. The reason was, that I wanted during the development to have the ability for taking a direct look on the old code from time to time when needed.

Before the final merge, i planed to make a whole cleanup to remove such parts. There are also functions and parameters which aren't used (anymore).